nixos/modules/services/networking/autossh.nix

```diff
@@ -0,0 +1,124 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.autossh;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.autossh = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable the AutoSSH, the SSH sessions manager
+        '';
+      };
```

Comment:
Isn't the `enable` flag superfluous? You should be able to use `cfg.sessions != []` instead.